### ISSUE TITLE
Session cannot be saved

### ISSUE THREAD

**Issue by jinmarcus** [bcit-ci/CodeIgniter4#958]
Session cannot be saved

session()->set('test', 123);

// Run to empty
echo session()->get('test');

**Comment by InsiteFX** [comment]
`$session->set('some_name', 'some_value');

Your calling it wrong...

**Comment by lonnieezell** [comment]
Nothing should have changed with the sessions recently. You should know it’s impossible for me to debug without more information up front. 

- What environment are you running under?
- what session driver are you using?
- what does your configuration look like?

And last but most definitely not least - what have you done to debug it yourself and show that it’s an actual bug and not just a misconfiguration!

**Comment by InsiteFX** [comment]
Lonnie,

Can we call it like that with the () or is that the helper way?

**Comment by lonnieezell** [comment]
Yes, it can be called that way. It's in the Common.php file, I believe where we provide a few helper methods for things like sesison, request, and response.

**Comment by InsiteFX** [comment]
Thank you for clarifying that for me.

**Comment by jinmarcus** [comment]
windows and centos7.0
php7.0 and php7.1
mysql
There's been a mistake.

**Comment by InsiteFX** [comment]
I'm on Windows 10 Pro I'll check it out today.

**Comment by lonnieezell** [comment]
@InsiteFX thanks for testing. 

@jinmarcus Can you verify your configuration works with, say the cache library using a Redis driver? At this point I can't confirm whether or not it's your system configuration (which is what I'm leaning toward at the moment) or an actual bug. and I know that Redis can be somewhat tricky to configure properly. Please check the docs, maybe head over to the phpredis page that's linked in the docs and do some experiments with the caching to see if you can narrow things down.

**Comment by JakeAi** [comment]
@lonnieezell A few weeks ago I was using Redis and it was working fine on Windows 10 Pro and 2 Instances of Server 2012.

**Comment by vallgorr** [comment]
I use DatabaseHandler and every time get 'new data'. Data in DB write and read correct. But something get wrong..
`		session()->start();
		if (session('logged_in') == 'true') {
			print_r(session()->get());
		} else {
			session()->set(array(
				'logged_in' => 'true',
			));
			print_r('new data');
		}`
![image](https://user-images.githubusercontent.com/7808206/37478246-5e52e322-2882-11e8-862c-ccc19cf97f24.png)
